pkg/engine/task/task_toggle.go

```diff
@@ -43,7 +43,13 @@ func (tt ToggleTask) delegateTask(ctx Context) (Tasker, error) {
 	if !exists {
 		return task, fmt.Errorf("no value for parameter %s found", tt.Parameter)
 	}
-	enabled, err := strconv.ParseBool(val)
+
+	stringVal, ok := val.(string)
+	if !ok {
+		return task, fmt.Errorf("value of parameter %s isn't a string", tt.Parameter)
```

Comment:
shouldn't there be an error on the use of a toggle param that is NOT a string... which is technically a bool ?
It seems like we are missing that..

Reply:
Isn't this what this check does? We try to cast to a string, if it's `!ok` we error out?

Reply:
Isn't that done by the line below? If `stringVal` is neither "true" nor "false", `strconv.ParseBool` will return an error.

Reply:
The error message is somewhat confusing, we expect a string/boolean there. Maybe smth. like:
```suggestion
		return task, fmt.Errorf("value of parameter %s must be either 'true' or 'false'", tt.Parameter)
```
wdyt?

Reply:
I disagree, because this is checked in the line below when parsing the string to bool. This line is checking that `typeof(interface) == string`, not the content of the string.

Reply:
If it's not a string it can't be `true` or `false` which is what we expect.

Reply:
Then "isn't a string representing a boolean" would be better, because booleans can also be represented by a boolean or integer type. Also `strconv.ParseBool` accepts 1, t, T, TRUE, true, True, 0, f, F, FALSE, false, False.